thunder/executors/nvfuserex_impl.py

```diff
@@ -464,9 +464,10 @@ class FusionDefinitionWrapper:
     disable_options: None | list[str] = None
 
     def __call__(self, *args):
-        fd = self.get_fd(self.to_descriptors(args))
-        self.last_used = fd
-
+        torch.cuda.nvtx.range_push("nvfuser_ex.__call__")
+        if self.last_used is None:
+            self.last_used = self.get_fd(self.to_descriptors(args))
+        fd = self.last_used
```

Comment:
IIRC, I think the reason the args are transformed to descriptors is that Thunder doesn't model properties of tensors, like strides, that nvFuser might use to pick among multiple file descriptors for the same fusion; maybe that's not the case anymore.

I'm not sure if we have any tests that check for calling the same fusion with tensors that pass Thunder's checks but have different strides. If not, we should probably add one

fyi @IvanYashchuk

Reply:
Unfortunately, as Mike says we can't generate the FusionDefinition once and use it for every later call. We need to be pessimistic here and expect that at runtime strides may change.
This type of execution should definitely be available as an option when we are sure strides will be the same for every invocation. It would also be great if nvFuser provided a way to directly execute the last compiled fusion bypassing all shape and stride checks.